docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+# Connection Tester
+
+The goal of this document is to sketch out possible approaches for implementing _Connection testing_ in the context of iPaaS. This backend concept is introduced to support the use case to quickly test whether a given `Connection`  (which is the combination of a `Connector` and a `ConnectorConfiguration`) is actually usable. The typical use case is, that the user pushes a button "Test connection" on the "Connection Details" page which actually triggers the test of the connection and that can be either successful or failing. When it fails, a detailed error message describes not only the error itself but also how to resolve the error (e.g. "Cannot connect because of invalid credentials"). 
+
+## Communication flow
+
+When the user triggers this test function from within the UI, a backend call to `ipaas-rest` will be issued, which in turn will trigger a (still to be defined) method on the referenced connector with the concrete configuration given. This method, let's call it `ping()`, must be side effect free, which should be possible for most endpoints. The method takes the configuration as argument for the connector and returns a result objects which contains the result of the check (ok / nok) and in case of a failure the error message (question: Should we already think about I18N here ?)
```

Comment:
I agree on side-effect free test first.  I like the idea of the more evolved test Kurt is mentioning, but I think there's some bleed over into actions with that test.  IOW, I'm connecting and performing an action - that's the output you want to capture.  As of now, actions are configured independent of connections (in integrations), so we need more discussion around where this would go.